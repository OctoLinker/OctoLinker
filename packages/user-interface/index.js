import './style.css';

export const showNotification = ({ body, type = 'info', id = 'common' }) => {
  const typeClass = {
    info: 'flash-info',
    warn: 'flash-warn',
    error: 'flash-error',
  };

  const el = document.createElement('div');
  el.innerHTML = `<div class="js-octolinker-flash flash flash-full ${typeClass[type]}">
    <div class="container">
      <button class="octolinker-flash-close flash-close js-flash-close js-flash-close-${id}" type="button" aria-label="Dismiss this message">
        <svg aria-hidden="true" class="octicon octicon-x" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"></path></svg>
      </button>
       <div class="octolinker-flash-image"></div>
       <div class="octolinker-flash-content">
        ${body}
       </div>
    </div>
  </div>`;

  document.getElementById('js-flash-container').append(el);

  return el;
};

export const removeAllNotifications = () => {
  document
    .querySelectorAll('.js-octolinker-flash')
    .forEach((el) => el.remove());
};

export const isPrivateRepository = () => !!document.querySelector('h1.private');
