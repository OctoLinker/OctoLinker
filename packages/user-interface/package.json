{
  "name": "@octolinker/user-interface",
  "version": "1.0.0",
  "description": "",
  "repository": "https://github.com/octolinker/octolinker/tree/main/packages/user-interface",
  "license": "MIT",
  "main": "./index.js"
}
