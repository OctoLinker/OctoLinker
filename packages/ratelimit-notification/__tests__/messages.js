import { showNotification } from '@octolinker/user-interface';
import {
  tokenIsInvalid,
  needsTokenForPrivate,
  rateLimitExceeded,
} from '../messages';

jest.mock('@octolinker/user-interface');

describe('ratelimit-notification', () => {
  it('tokenIsInvalid', () => {
    const showNotificationStub = jest.fn();
    showNotification.mockImplementation(showNotificationStub);

    tokenIsInvalid();
    expect(showNotificationStub.mock.calls[0][0]).toMatchSnapshot();
  });
  it('needsTokenForPrivate', () => {
    const showNotificationStub = jest.fn();
    showNotification.mockImplementation(showNotificationStub);

    needsTokenForPrivate();
    expect(showNotificationStub.mock.calls[0][0]).toMatchSnapshot();
  });

  describe('rateLimitExceeded', () => {
    it('for authenticated requests', () => {
      const showNotificationStub = jest.fn();
      showNotification.mockImplementation(showNotificationStub);

      rateLimitExceeded({ isUnauthenticated: false, remainingTime: 5000 });
      expect(showNotificationStub.mock.calls[0][0]).toMatchSnapshot();
    });

    it('for unauthenticated requests', () => {
      const showNotificationStub = jest.fn();
      showNotification.mockImplementation(showNotificationStub);

      rateLimitExceeded({ isUnauthenticated: true, remainingTime: 300000 });
      expect(showNotificationStub.mock.calls[0][0]).toMatchSnapshot();
    });
  });
});
