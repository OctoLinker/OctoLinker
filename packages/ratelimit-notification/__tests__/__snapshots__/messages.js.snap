// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ratelimit-notification needsTokenForPrivate 1`] = `
Object {
  "body": "OctoLinker needs a GitHub API token to retrieve repository metadata for private repositories. <a href=\\"#\\" class=\\"js-octolinker-open-settings\\">Create a token</a> to enable OctoLinker for all your private repositories. <a href=\\"#\\" class=\\"js-octolinker-open-settings\\">Disable OctoLinker for private repositories</a> to get rid of this notification.",
  "clickHandler": [Function],
  "type": "info",
}
`;

exports[`ratelimit-notification rateLimitExceeded for authenticated requests 1`] = `
Object {
  "body": "OctoLinker exceed the GitHub API hourly limit. The rate limit will be reset in 5 seconds.",
  "clickHandler": [Function],
  "type": "warn",
}
`;

exports[`ratelimit-notification rateLimitExceeded for unauthenticated requests 1`] = `
Object {
  "body": "OctoLinker exceed the GitHub API hourly limit for unauthenticated requests. You probably want to <a href=\\"#\\" class=\\"js-octolinker-open-settings\\">create a token</a> or wait 5 minutes.",
  "clickHandler": [Function],
  "type": "warn",
}
`;

exports[`ratelimit-notification tokenIsInvalid 1`] = `
Object {
  "body": "The token you provided is invalid. You must <a href=\\"#\\" class=\\"js-octolinker-open-settings\\">create a new token</a> before you can continue using OctoLinker.",
  "clickHandler": [Function],
  "type": "error",
}
`;
